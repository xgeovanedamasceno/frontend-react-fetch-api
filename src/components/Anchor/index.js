import PropTypes from 'prop-types';
import AnchorStyle from './styled';

function Anchor({ url, name }) {
  return (
    <AnchorStyle>
      <a href={url}>{ name }</a>
    </AnchorStyle>
  );
}

Anchor.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

export default Anchor;
