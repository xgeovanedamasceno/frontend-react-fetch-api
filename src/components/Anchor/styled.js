import styled from 'styled-components';

const AnchorStyle = styled.span`
  background-color: #F6F8FA;
  border-radius: 5px;
  color: #24292F;
  padding: 16px 8px;
`;

export default AnchorStyle;
