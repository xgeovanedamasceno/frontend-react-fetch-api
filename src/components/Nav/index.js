import PropTypes from 'prop-types';
/* import Anchor from '../Anchor'; */

function Nav({ children }) {
  return (
    <nav>
      { children }
    </nav>
  );
}

Nav.propTypes = {
  children: PropTypes.element.isRequired
};

export default Nav;
