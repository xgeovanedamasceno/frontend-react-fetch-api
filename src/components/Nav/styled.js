import styled from 'styled-components';

const NavStyle = styled.span`
  margin: 0 auto;
  width: 750px;

`;

export default NavStyle;
