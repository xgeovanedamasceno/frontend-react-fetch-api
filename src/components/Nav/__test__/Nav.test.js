import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Nav from '..';
import Anchor from '../../Anchor';

describe('Nav', () => {
  it('should render an nav element with 1 anchor component', () => {
    render(
      <Nav>
        <Anchor url="#exemplo" name="Products" />
      </Nav>
    );

    const links = screen.getByRole('link', { name: 'Products' });

    const navEl = screen.getByRole('navigation');

    expect(navEl).toContainElement(links);
  });

  /* it('should render an nav element with 2 anchor components', () => {
    render(
      <Nav>
        <Anchor url="#exemplo" name="Products" />
        <Anchor url="#exemplo" name="Productz" />
      </Nav>
    );

    const links = screen.getAllByRole('link');

    const navEl = screen.getByRole('navigation');
    expect(links).toHaveLength(1);
  }); */

  it('should render correctly', () => {
    const tree = renderer.create(
      <Nav>
        <Anchor url="#exemplo" name="Products" />
      </Nav>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
