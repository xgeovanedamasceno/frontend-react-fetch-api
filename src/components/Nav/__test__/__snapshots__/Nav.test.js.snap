// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render correctly 1`] = `
.c0 {
  background-color: #F6F8FA;
  border-radius: 5px;
  color: #24292F;
  padding: 16px 8px;
}

<nav>
  <span
    className="c0"
  >
    <a
      href="#exemplo"
    >
      Products
    </a>
  </span>
</nav>
`;
